daras_ai_v2/language_model.py

```diff
@@ -208,20 +213,23 @@ class LargeLanguageModels(Enum):
         llm_api=LLMApis.groq,
         context_window=32_768,
         price=1,
+        supports_json=True,
     )
     gemma_2_9b_it = LLMSpec(
         label="Gemma 2 9B (Google)",
         model_id="gemma2-9b-it",
         llm_api=LLMApis.groq,
         context_window=8_192,
         price=1,
+        supports_json=True,
     )
     gemma_7b_it = LLMSpec(
         label="Gemma 7B (Google)",
         model_id="gemma-7b-it",
         llm_api=LLMApis.groq,
         context_window=8_192,
         price=1,
+        supports_json=True,
```

Comment:
Shouldn’t this include validation and retries to make sure the output is JSON similar to the openai implementation?

Reply:
groq supports json mode

Reply:
https://console.groq.com/docs/text-chat#json-mode-object-object